### PR TITLE
ZCS-11624 Delegated Admin- Assign default domain administrator views and rights checkbox not getting saved

### PR DESCRIPTION
**Pull request by subrahmanyaShetty090**
Making "GetGrantsRequest" API call to get the grants assigned to a particular account
Setting the value of "isAssignDefaultDARights" to "TRUE" if all the 3 rights have been assigned to that account

### CODE REVIEW

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js [hunk 1]
```diff
@@ -1721,6 +1731,23 @@ function(by, val) {
...
+        if(allGrantsList[i].grantee_id === this.id){
+            if((arr.includes(allGrantsList[i].right)) && (allGrantsList[i].deny === "0")){
+                count++;
+            }
+        }
+        
+    }
+}else{
+    count=0;
+}
```
Comment (lavegupta) [inline WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js:1747]: indent

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js [hunk 2]
```diff
@@ -1721,6 +1731,23 @@ function(by, val) {
     } else {
         this[ZaAccount.A2_autodisplayname] = "FALSE";
     }
+    let arr=["domainAdminZimletRights","adminLoginCalendarResourceAs","domainAdminConsoleRights"];
```
Comment (lavegupta) [inline WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js:1734]: Don't use ES6 Features.

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js [hunk 3]
```diff
@@ -1721,6 +1731,23 @@ function(by, val) {
...
+    var count=0;
+    if( allGrantsList.length >= 3){
+    for(var i=0; i< allGrantsList.length; i++){
+        if(allGrantsList[i].grantee_id === this.id){
+            if((arr.includes(allGrantsList[i].right)) && (allGrantsList[i].deny === "0")){
+                count++;
+            }
+        }
```
Comment (lavegupta) [inline WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js:1742]:
```
if(allGrantsList[i].grantee_id === this.id && arr.includes(allGrantsList[i].right) && allGrantsList[i].deny === "0"){
 count++;
}
```

**Review by k-kato — CHANGES_REQUESTED** [review]
all code related to delegated admin management should be moved to private repo.

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js [hunk 4]
```diff
@@ -1721,6 +1730,29 @@ function(by, val) {
     } else {
         this[ZaAccount.A2_autodisplayname] = "FALSE";
     }
+       
+    var defaultRights = ZaNewAdminWizard.getDefaultDARights (this) ;
+    var count=0;
+    if (allGrantsList.length >= 3) {
+      for (var i = 0; i < allGrantsList.length; i++) {
+        if (allGrantsList[i].grantee_id === this.id) {
```
Comment (lavegupta) [inline WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js:1738]:
```
if (count > 2) { break; }
if (allGrantsList[i].grantee_id === this.id) {
```

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js (hunk 3 above)
Comment (subrahmanyaShetty090) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js]: Moved these changes to https://github.com/Zimbra/zm-delegated-admin-zimlet/pull/8

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js (hunk 2 above)
Comment (subrahmanyaShetty090) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js]: moved the changes to https://github.com/Zimbra/zm-delegated-admin-zimlet/pull/8

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js (hunk 1 above)
Comment (subrahmanyaShetty090) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js]: moved these changes to https://github.com/Zimbra/zm-delegated-admin-zimlet/pull/8

File: WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js (hunk 4 above)
Comment (subrahmanyaShetty090) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js]: Moved these changes to https://github.com/Zimbra/zm-delegated-admin-zimlet/pull/8

Comment (silentsakky) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js]: Please close this PR and delete branch

Comment (subrahmanyaShetty090) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js]: closing the PR

**Pull request closed without merging**